src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -49,128 +51,276 @@ internal class CompletionItem : IComparable<CompletionItem>
         public string SortText { get; }
 
         /// <summary>
-        /// Whether or not this item should be preselected when presented to the user.  It is up to
-        /// the ICompletionRules to determine how this flag should be handled.  However, the default
-        /// behavior is that, if there has been no filter text then a preselected item is preferred
-        /// over any other item. If there has been filter text supplied, then a preselected item is
-        /// preferred over another item if the ICompletionRules currently in effect deem them
-        /// otherwise identical.
+        /// Whether or not this item should be preselected when presented to the user.
         /// </summary>
         public bool Preselect { get; }
 
         /// <summary>
-        /// The span(respective to the original document text when this completion item was created)
-        /// to use for determining what text should be used to filter this completion item against.
+        /// The span of the item with respect to the original document text.
+        /// Helps determine what text should be used to filter this completion item against.
+        /// </summary>
+        /// <remarks>
         /// Most commonly this is the same text span that is in TextChange, however in specialized
         /// cases it can be different.  For example, in C#, if a user types "foo." the item "operator
-        /// int" may be placed in the list.  It's filter span will be created a after the dot
+        /// int" may be placed in the list.  It's filter span will be created after the dot
         /// position (so that typing "oper" will help filter down to the list of operators).
         /// However, the text change may extend further backward so that if that item is committed
-        /// the resultant text becomes "((int)foo).
-        /// </summary>
+        /// the resultant text becomes "((int)foo).</remarks>
         public TextSpan FilterSpan { get; }
 
         /// <summary>
-        /// A CompletionItem marked as a builder will be presented as an Intellisense Builder,
-        /// initially with its display text, which will be replaced as the user types.
+        /// The description of the item broken down into display parts.
+        /// If the description is empty, then the description can be accessed via <see cref="CompletionService.GetDescriptionAsync(Document, CompletionItem, CancellationToken)"/>.
+        /// </summary>
+        public ImmutableArray<SymbolDisplayPart> Description { get; }
```

Comment:
Ideally removed.  I'd like to knwo the cases where we think this is more appropriate/necessary.